src/devices/video/mc6845.cpp

```diff
@@ -416,9 +416,9 @@ void mc6845_device::recompute_parameters(bool postload)
 	uint16_t horiz_pix_total = (m_horiz_char_total + 1) * m_hpixels_per_column;
 	uint16_t vert_pix_total = (m_vert_char_total + 1) * video_char_height + m_vert_total_adj;
 
-	/* determine the visible area, avoid division by 0 */
-	uint16_t max_visible_x = m_horiz_disp * m_hpixels_per_column - 1;
-	uint16_t max_visible_y = m_vert_disp * video_char_height - 1;
+	// properly handle horizonal display of 0
+	uint16_t max_visible_x = m_horiz_disp ? m_horiz_disp * m_hpixels_per_column - 1 : 0;
+	uint16_t max_visible_y = m_vert_disp ? m_vert_disp * video_char_height - 1 : 0;
```

Comment:
Ideally, it would be 0 pixels, but since the variable is unsigned, having a max value of 0 is the lowest it can be set to. With the prior calculation if m_vert_disp is 0, the max becomes 0xffff. 
This causes the check on line 453 to consider that the crtc params are invalid, and will not update the screen. So when the ROM sets it to 0, to trigger the screen saver, the screen is frozen, instead of being cleared. If there is a better way to handle this, let me know.

Reply:
You’re going to need to rework the code so that `max_visible_x`, `max_visible_y`, `m_max_visible_x` and `m_max_visible_y` are exclusive rather than inclusive (probably a good idea to rename them to something like `visible_width`/`visible_height` as the semantics are changing).  That will allow it to deal with the zero width/height.

I don’t think changing the screen device’s visible area to a single row/column is a great idea, either.  The system is only going to do this as a “blank screen” trick, and it will most likely program the same visible area again afterwards.

I think it would be better to detect zero width or height, and when it happens, try to leave the visible area unchanged if possible (you may need to limit it to the total width/height if they’re reduced while zero width and/or height is set).  That would give a better user experience.

Reply:
Variables have been updated. I didn't see any user experience issues when the screen saver activated with the current code. 

Since the calculation is no longer setting the variable values to 0xffff, which had previously preventing calculating the resolution at startup based on the class initialized values. I had to set `m_horiz_char_total` and `m_vert_char_total` to be less than `m_horiz_disp` and `m_vert_disp` to prevent the startup resolution screen to show bad values.